commands.go

```diff
@@ -224,15 +234,27 @@ func runCommands(line string, u *user, isSlack bool) {
 				roomsInfo += fmt.Sprintf("%s: %d  \n", blue.Paint(kv.roomName), kv.numOfUsers)
 			}
 			b("", "Rooms and users  \n"+strings.TrimSpace(roomsInfo))
-		}
-		if strings.HasPrefix(rest, "#") {
+		} else if strings.HasPrefix(rest, "#") {
 			//rest = strings.TrimSpace(strings.TrimPrefix(line, "#"))
 			if v, ok := rooms[rest]; ok {
 				u.changeRoom(v, sendToSlack)
 			} else {
 				rooms[rest] = &room{rest, make([]*user, 0, 10), sync.Mutex{}}
 				u.changeRoom(rooms[rest], sendToSlack)
 			}
+		} else if strings.HasPrefix(rest, "@") {
+			restSplit := strings.Fields(strings.TrimPrefix(rest, "@"))
+			peer, ok := findUserByName(u.room, restSplit[0])
+			if !ok {
+				u.writeln(devbot, "No such person lol, who you wanna dm? (you might be in the wrong room)")
+				return
+			}
+			u.messaging = peer
+			u.writeln(devbot, fmt.Sprintf("Now messaging %s. To leave say \n>./room", peer.name))
+			return
+		} else {
```

Comment:
Don't have an else statement since this will catch on to any message if I understand correctly

Reply:
Nope, I did not understand correctly